src/app/components/shared/input/range/range.component.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  input,
+  output,
+  signal,
+} from "@angular/core";
+import { ControlValueAccessor, FormsModule } from "@angular/forms";
+import { toNumber } from "@helpers/typing/toNumber";
+
+@Component({
+  selector: "baw-range",
+  templateUrl: "./range.component.html",
+  styleUrl: "./range.component.scss",
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [FormsModule],
+})
+export class RangeComponent implements ControlValueAccessor {
+  public label = input("");
+  public min = input(0);
+  public max = input(100);
+  public step = input(5);
+
+  public value = signal<number | null>(0);
+  protected disabled = signal(false);
+
+  private onChange: (value: number | null) => void;
+  private onTouched: () => void;
+
+  // We want to output the "input" event so that this component can be used as
+  // a direct replacement for the input:range element.
+  // By emitting the "input" event, we can also seamlessly use this component
+  // with directives that attach to input elements e.g. the "bawDebouncedInput"
+  // directive.
+  //
+  // eslint-disable-next-line @angular-eslint/no-output-native
+  public input = output<number>();
+
+  public writeValue(value: any): void {
+    this.value.set(value);
+    this.onChange?.(this.value());
+  }
+
+  public registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  public registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  public setDisabledState?(isDisabled: boolean): void {
+    this.disabled.set(isDisabled);
+  }
+
+  protected updateValue(event: Event) {
+    if (!(event.target instanceof HTMLInputElement)) {
+      return;
+    }
```

Comment:
is this guard needed? are you binding the handler elsewhere?

Reply:
No. This is just a type guard with more safety

perf is negligible because JIT can branch predict it quite easily

Reply:
perf argument: accepted

semantics were what lead me to query this though.

Reply:
![image](https://github.com/user-attachments/assets/177f29de-0130-461d-967a-275e4e2bb492)

I'm going to use type casts from now on. It's more descriptive and easier to read